### ISSUE TITLE
Sentence Transformer encodings

### ISSUE THREAD

**Issue by Gchand0249** [UKPLab/sentence-transformers#1168]
Hi , 

We finetuned Sentence transformers on our domain specific data. It is giving high cosine score for irrelevant suggestions . We used good , bad , ok while labeling the data. We have around 100k data.

**Comment by nreimers** [comment]
Adding a question to your issue would be quite helpful.

**Comment by Gchand0249** [comment]
yes , After extracting embeddings from sbert we using cosine score for sorting results. Issue here is results with high cosine score are irrelevant . And similar results are getting less score. We are unable to figure it out why it is happening

**Comment by nreimers** [comment]
Likely due to wrongly training the model.

**Comment by Gchand0249** [comment]
Thank you , Does performance depends on batch size ?

**Comment by nreimers** [comment]
SoftmaxLoss is the wrong loss. Have a look at the other losses functions

**Comment by Gchand0249** [comment]
Thanks for your replay , Could you please suggest me preferable loss to train sbert with good, bad ,okay labels ?

**Comment by nreimers** [comment]
MultipleNegativesRankingLoss or one of the triplet losses

**Comment by Gchand0249** [comment]
Thank you @nreimers , Do you prefer MultipleNegativesRankingLoss or one of the triplet losses for cross-encoder also ?

**Comment by Gchand0249** [comment]
Thank you @nreimers ,

Could you please explain why SoftmaxLoss is the wrong loss ? In the sbert website you mentioned that you used softmax loss for training sbert on NLI data and our data labels are similar to NLI data.

**Comment by nreimers** [comment]
That it works on NLI is rather a coincidence, but there is not good logic behind it:
https://www.sbert.net/examples/training/nli/README.html#multiplenegativesrankingloss

**Comment by Gchand0249** [comment]
Thank you for you suggestion ,  In hard multiplenegativesrankingloss , team stated  that  **"" You can also provide one or multiple hard negatives per anchor-positive pair by structering the data like this: (a_1, p_1, n_1), (a_2, p_2, n_2)
Here, n_1 is a hard negative for (a_1, p_1). The loss will use for the pair (a_i, p_i) all p_j (j!=i) and all n_j as negatives. ""**


Could you please eloberate this statement . Does it mean loss use P_j and n_j as negatives for a_i ?

**Comment by nreimers** [comment]
Yes

**Comment by Gchand0249** [comment]
We did synonym expansion for data and  in our case most of a_i and p_j  are positive. What can we do in this case ?

**Comment by Gchand0249** [comment]
We did synonym expansion for data and in our case most of a_i and p_j are positive. How does it works in this case. Won't it effect the embeddings?

**Comment by nreimers** [comment]
Then you have to create a custom DataLoader that ensures that a batch does not contain two entries of the same type

**Comment by Gchand0249** [comment]
It is very difficult for us to extract two entries of the same type. Is it okay to go with triplet loss ?

**Comment by nreimers** [comment]
Sure

**Comment by Gchand0249** [comment]
Thank you .... Does distance_metric in triplet loss has any impact on performance ? We tried with default Euclidean preformance was not good so we are trying now with COsine.
